src/screens/ScanPreviewScreen.tsx

```diff
@@ -0,0 +1,282 @@
+import { useActionSheet } from '@expo/react-native-action-sheet';
+import { NativeStackScreenProps } from '@react-navigation/native-stack';
+import React, { useCallback, useEffect, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Image, ImageBackground, ScrollView, StyleSheet, TouchableOpacity, View } from 'react-native';
+import DocumentScanner, { ResponseType, ScanDocumentResponseStatus } from 'react-native-document-scanner-plugin';
+import * as Sentry from 'sentry-expo';
+
+// @ts-ignore
+import DefaultImagePreviewBackground from '../../assets/default-image-preview-background.png';
+import SvgArrowLeftIcon from '../../assets/icons/arrow-round-left.svg';
+import SvgEllipsisIcon from '../../assets/icons/ellipsis-icon.svg';
+import SvgPlusIcon from '../../assets/icons/plus-icon.svg';
+import SvgTrashIcon from '../../assets/icons/trash-icon.svg';
+import SvgUploadIcon from '../../assets/icons/upload-icon.svg';
+import Text from '../components/Text/Text';
+import { useGetCurrentCustomer } from '../hooks/queries/useGetCurrentCustomer';
+import { useGetCurrentUser } from '../hooks/queries/useGetCurrentUser';
+import { RootStackParamList } from '../navigation/types';
+import { useImageStore } from '../store/ImageStore';
+import { colors, dimensions } from '../theme/';
+
+type Props = NativeStackScreenProps<RootStackParamList, 'ScanPreviewScreen'>;
+
+export const ScanPreviewScreen: React.FC<Props> = ({ navigation }) => {
+  const { showActionSheetWithOptions } = useActionSheet();
+  const currentCustomer = useGetCurrentCustomer();
+  const { currentUser } = useGetCurrentUser();
+  const {
+    addImages,
+    company,
+    deleteImage,
+    documentType,
+    images,
+    reset,
+    selectedImageIndex,
+    selectImage,
+  } = useImageStore();
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    if (company === undefined) {
+      navigation.navigate('ScanSelectCompanyScreen');
+    } else if (documentType === undefined) {
+      navigation.navigate('ScanSelectDocumentTypeScreen');
+    }
+  }, [company, documentType, navigation]);
+
+  const isUploadEnabled = useMemo(() => images.length > 0, [images]);
+
+  const openCamera = useCallback(() => {
+    DocumentScanner.scanDocument({
+      croppedImageQuality: 100,
+      letUserAdjustCrop: true,
+      responseType: ResponseType.ImageFilePath,
+    })
+      .then((result) => {
+        // If the status is not Success, the user cancelled
+        if (result.status === ScanDocumentResponseStatus.Success) {
+          addImages(result.scannedImages || []);
+        }
+      })
+      .catch(() => {
+        Sentry.Native.captureMessage('Error occurred while scanning documents', 'error');
```

Comment:
Misschien ook de error terugsturen wat in catch voorkomt. Maakt het debuggen iets duidelijker.

Reply:
Idem opgelost als in `ImageStore`